src/flag_gems/ops/cumsum.py

```diff
@@ -48,6 +49,65 @@ def cumsum_kernel(
     tl.store(out_ptrs, result, mask=mask)
 
 
+
+@libentry()
+@triton.autotune(
+    configs=[
+        triton.Config({"BLOCK_M": 8, "BLOCK_N": MAX_C_MLU_CUMSUM//4}, num_warps=8, num_stages=4),
+        triton.Config({"BLOCK_M": 8, "BLOCK_N": MAX_C_MLU_CUMSUM//4}, num_warps=8, num_stages=5),
+        triton.Config({"BLOCK_M": 16, "BLOCK_N": MAX_C_MLU_CUMSUM//4}, num_warps=8, num_stages=4),
+        triton.Config({"BLOCK_M": 16, "BLOCK_N": MAX_C_MLU_CUMSUM//4}, num_warps=8, num_stages=5),
+        triton.Config({"BLOCK_M": 32, "BLOCK_N": MAX_C_MLU_CUMSUM//4}, num_warps=8, num_stages=4),
+        triton.Config({"BLOCK_M": 32, "BLOCK_N": MAX_C_MLU_CUMSUM//4}, num_warps=8, num_stages=5),
+        triton.Config({"BLOCK_M": 8, "BLOCK_N": MAX_C_MLU_CUMSUM//2}, num_warps=8, num_stages=4),
+        triton.Config({"BLOCK_M": 8, "BLOCK_N": MAX_C_MLU_CUMSUM//2}, num_warps=8, num_stages=5),
+        triton.Config({"BLOCK_M": 16, "BLOCK_N": MAX_C_MLU_CUMSUM//2}, num_warps=8, num_stages=4),
+        triton.Config({"BLOCK_M": 16, "BLOCK_N": MAX_C_MLU_CUMSUM//2}, num_warps=8, num_stages=5),
+        triton.Config({"BLOCK_M": 32, "BLOCK_N": MAX_C_MLU_CUMSUM//2}, num_warps=8, num_stages=4),
+        triton.Config({"BLOCK_M": 32, "BLOCK_N": MAX_C_MLU_CUMSUM//2}, num_warps=8, num_stages=5),
+        triton.Config({"BLOCK_M": 8, "BLOCK_N": MAX_C_MLU_CUMSUM}, num_warps=8, num_stages=4),
+        triton.Config({"BLOCK_M": 8, "BLOCK_N": MAX_C_MLU_CUMSUM}, num_warps=8, num_stages=5),
+        triton.Config({"BLOCK_M": 16, "BLOCK_N": MAX_C_MLU_CUMSUM}, num_warps=8, num_stages=4),
+        triton.Config({"BLOCK_M": 16, "BLOCK_N": MAX_C_MLU_CUMSUM}, num_warps=8, num_stages=5),
+        triton.Config({"BLOCK_M": 32, "BLOCK_N": MAX_C_MLU_CUMSUM}, num_warps=8, num_stages=4),
+        triton.Config({"BLOCK_M": 32, "BLOCK_N": MAX_C_MLU_CUMSUM}, num_warps=8, num_stages=5),
+    ],
+    key=[
+        "M",
+        "N",
+    ],
+)
```

Comment:
Using only M & N in tunning key while M, N &K may all affect the performance would cause some unexpected behavior. For example, the best config tunned for (m, n, k1) may not be the same config tunned from (m, n, k2), so previous runs have effects on peformance afterwards.